demos/pkcs11/pkcs11_demo_management_and_rng/pkcs11_demo_management_and_rng.c

```diff
@@ -0,0 +1,238 @@
+/*
+ * FreeRTOS PKCS #11 V1.0.3
+ * Copyright (C) 2020 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *
+ * http://aws.amazon.com/freertos
+ * http://www.FreeRTOS.org
+ */
+
+/* Standard includes. */
+#include <stdio.h>
+#include <stdlib.h>
+#include <assert.h>
+
+/* Logging configuration for the PKCS #11 library. */
+#ifndef LIBRARY_LOG_NAME
+    #define LIBRARY_LOG_NAME    "PKCS11_DEMO"
+#endif
+
+#ifndef LIBRARY_LOG_LEVEL
+    #define LIBRARY_LOG_LEVEL    LOG_INFO
+#endif
+
+#include "logging_stack.h"
+
+/* PKCS #11 includes. */
+#include "core_pkcs11.h"
+#include "pkcs11.h"
+
+/**
+ * @brief This function details how to use the PKCS #11 "Management" functions to
+ * manage the internal state machine of the PKCS #11 implementation. These
+ * functions are all defined in
+ * http://docs.oasis-open.org/pkcs11/pkcs11-base/v2.40/os/pkcs11-base-v2.40-os.html
+ * please consult the standard for more information regarding these functions.
+ *
+ * The standard has grouped the functions presented in this demo as:
+ * General Purpose Functions
+ * Slot and Token Management Functions
+ * Session Management Functions
+ * Random Number Generation Functions
+ */
+void vPKCS11ManagementAndRNGDemo( void )
+{
+    /* We will use the terminology as defined in the standard, Cryptoki is in
+     * reference to the Cryptographic Token Interface defined in the PKCS #11
+     * standard. An implementation of Cryptoki is referred to as a
+     * "Cryptoki library". */
+    LogInfo( ( "Starting PKCS #11 Management and Random Number Generation" \
+               " Demo." ) );
+
+    /* CK_RV is the return type for a Cryptoki function. Generally the underlying
+     * type is a CK_ULONG, it can also be a CKR_VENDOR_DEFINED type. */
+    CK_RV xResult = CKR_OK;
```

Comment:
Hungarian notation of variable naming is not used in C-SDK. (Demo and test wide changes can be made in a separate PR though)

Reply:
Hmm okay, that will be a big change so agreed, separate PR will be best